### DISCUSSION TITLE
How to use the dl module to call WinExec to open the calculator?

### DISCUSSION THREAD

**Discussion by sonwnja** [vlang/v#8573] [General]
I want to use the WinExec function of kerenl32.dll to pop up the calculator

I tried the WinExec function of kerenl32.dll to pop up the calculator, but the calculator did not pop up and there was no error message. My code is as follows:
```V
module main

import dl

type Exec = fn (string, int) int

fn main() {
	h := dl.open('kernel32', dl.rtld_now)
	mut proc := &Exec(0)
	proc = dl.sym(h, 'WinExec')
	res := proc("calc.exe",2)
	println(res)
}
```
The results are as follows:
![image](https://user-images.githubusercontent.com/15842446/107009068-8019d580-67cf-11eb-9a1d-0c1b371b1963.png)

**Comment by JalonSolov** [comment]
There are 3 cross-platform routines in the standard library for starting other commands/processes:
```v
os.exec
os.new_process
os.system
```
Perhaps one of these will do what you need, without having to figure out how to call the Windows API directly?

**Reply by sonwnja** [reply]
I don’t want to use V to run the calculator. I just want to see if V can use the API functions in kernel32.dll or user32.dll. Can someone tell me how to call WinExec, or other API functions?

**Comment by JalonSolov** [comment]
Just look through the V source for files with `_windows` as part of the name.  You'll see examples of calling API functions in them.

2 in specific I know of are `vlib/clipboard/clipboard_windows.c..v` and `vlib/os/os_windows.c.v`.  There are plenty more.

You can also look in other files for `$if windows` for Windows-specific code.
